### PR TITLE
add quick start

### LINKED ISSUE #3651: napari for beginners page
## 📚 Documentation

Goal: 
Create a page for people to know what napari does and have the first try efficiently.

Target audience:
- Know nothing about napari
- May or may not know Python
- Interested in image visualization and/or analysis

Possible contents:
- [ ] 3 minutes "napari overview" video
- [ ] Bullet point of main features
- [ ] Quick start tutorial : installation (viewer + plugins), open sample image, visualization, analysis, export
- [ ] "Next steps" with hyperlinks to other materials 

Dependencies:
- [ ] bundled app?

@melissawm @DragaDoncila @GenevieveBuckley @neuromusic @dgmccart @tdimino @liaprins-czi

### PR DESCRIPTION
**Pull request by chili-chiu**
# Description

Add Quick Start as a place for new comers to try napari and have a glance of what it does.

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

closes #3651 

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: maybe remove "provide" in second the "provide layers" clause?

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: if we're targeting newcomers, we should explain conda env creation from the installation guide.  Since installing into an existing env is one of the more common sources of bugs.  for example, `napari[all]` will install pyqt5 via pip, which will very likely break their environment if they already have pyqt installed via conda

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]:
```suggestion
napari can be opened in one of [multiple ways](https://napari.org/tutorials/fundamentals/getting_started.html), depending on how it's used in your image analysis workflow.
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]:
```suggestion
The layer controls panel at the upper left of the viewer allows you to adjust contrast, colormap, and other layer properties. These settings affect the visualization, but do not affect the underlying data.
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]:
```suggestion
To change the image display through the [API](https://napari.org/api/stable/napari.layers.Image.html), in IPython console, type
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]:
```suggestion
To measure the area of the cell, we can use a labels layer and manually "paint" the cell.
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: I believe you can do a markdown numbered list with:

```md
1. Add a new labels layer
2. Click on "paint"
...
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: if you want to force a new line in markdown, use an empty line rather than `<br>`

```md
first line

new line
```

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: might remove the line "though the increased dimension could add to overall complexity."  (not wrong, just comes off a little odd)

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]: I almost want to clarify here that a plugin isn't necessary to totally customize one's own napari experience, but more when you're ready to share something that's working well for you with others.  I do worry that the napari plugin ecosystem might quick get inundated with people casually trying out (and posting to pypi) a plugin, simply because they thought it was the only way to add a widget to napari, for example.

File: docs/quick_start.md
Comment (tlambert03) [inline docs/quick_start.md]:
```suggestion
- Help the napari project by [filing issues](https://github.com/napari/napari/issues) for bugs you encounter or features that you'd like to see! Or even better, come [join the community meeting](https://napari.org/community/meeting_schedule.html) and get to know the team.
```

**Review by tlambert03 — COMMENTED** [review]
awesome!  a great thing to have in the docs. left only minor comments

File: docs/quick_start.md
Comment (chili-chiu) [reply docs/quick_start.md]: Good point. What about 

- If you don't see the functions you need in existing plugins, try to make your own [widget] (hyperlink), and publish it as a [plugin] (hyperlink) if you'd like to share with the community.

Comment (tlambert03) [reply docs/quick_start.md]: yeah, something like that.  Maybe even two sentences... so rather than "..., and publish it if you'd like to share", have a new sentence like "If you've develop some functionality that you think would benefit the community as a whole, consider publishing it as a plugin"

File: docs/quick_start.md
Comment (psobolewskiPhD) [inline docs/quick_start.md]: This is a bit tricky! `aicsimageio` is *awesome* but just installing the plugin `napari-aicsimageio` using the napari GUI will likely not provide czi, lif, or nd2 reading. It *will* enable out-of-memory reading though, so it's still sweet, but to get those formats you need to install extra things using pip. Further, in the case of CZI the key lib is compiled so it may not install on all platforms.

File: docs/quick_start.md
Comment (chili-chiu) [reply docs/quick_start.md]: Yes, I will include conda env creation steps here :)

I'm also working on another PR to update the installation and getting started pages. Right now we mix Python package installation and bundled app installation together, which can be quite confusing. 
By separating these two paths, it should be clearer what prerequisites and usage cases are specific to each.

File: docs/quick_start.md
Comment (psobolewskiPhD) [inline docs/quick_start.md]: Is the full scikit-image always installed with napari? I know sample data is, but does that ensure the user has `regionprops`?
Even if it is, it may be worth mentioning this is a different (and widely used) python module (library?). For someone new to the ecosystem—e.g. me 6 mo ago—this may not be obvious.

File: docs/quick_start.md
Comment (chili-chiu) [reply docs/quick_start.md]: @psobolewskiPhD is there a reader plugin(s) that you use regularly and think would be a good example instead of aicsimageio?

Comment (psobolewskiPhD) [reply docs/quick_start.md]: AICSImageIO was one of the things that moved me to napari and python... 
It's the one I use—but it requires a bit extra on the install side.
There was some idea of changing the license on it so it would be able to bundle the proprietary types in the napari plugin for ease of (GUI) use. 
Perhaps @JacksonMaxfield can chime in?

Comment (evamaxfield) [reply docs/quick_start.md]: Hey all! I would _love_ to see this recommendation make it in personally. The holdup has honestly been me -- grad school and other projects have left me with little time :sweat_smile:

That said, iirc, we got approval from allen legal to move napari-aicsimageio over to GPL license and bundle all the various reader into the dependencies. So I will try to get to that this week and make a new release that fixes that. The recommendation here then would be: "napari-aicsimageio is good for lots of different file formats but comes with a GPL license"? Idk, it seems weird to include but hard to say how much of napari-aicsimageio is used for downstream processing / analysis and not only viewing. Like will napari-aicsimageio taint the rest of the workflow constructed by a scientist with a GPL license?

Comment (psobolewskiPhD) [reply docs/quick_start.md]: The other option is that napari plugins can somehow communicate extra depends to the user. I've played around with ipython magics (https://github.com/napari/napari/discussions/3899#discussioncomment-1894365)
and `%pip install readlif` should JustWork in the napari console. the CZI one is compiled, so it's trickier, not sure about nd2...

File: docs/quick_start.md
Comment (shahidhaider) [reply docs/quick_start.md]: In fact, if you use 
1.
1.
1.
in markdown, it will auto populate the numbers and makes later insertion really easy!

File: docs/quick_start.md
Comment (evamaxfield) [reply docs/quick_start.md]: Update: I hope to have a napari-aicsimageio patch out next week. Sorry for the delay but we have actually had development progress hahaha

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]: I like to liberally add links everywhere but they are not strictly necessary :)
```suggestion
napari is a fast, interactive, multi-dimensional image viewer, with [a vibrant plugin ecosystem](https://www.napari-hub.org/) that expands its capability to tackle various domain-specific visualization and analysis needs. It is built on Qt (for the GUI), [vispy](https://vispy.org/) (for performant GPU-based rendering), and the scientific Python stack (numpy, scipy, and scikit-image). 
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]: I suggest using internal links whenever possible - especially to make sure that local docs build are successful and as complete as possible.
```suggestion
At its core, it provides critical viewer features out-of-the-box, such as support for [large multi-dimensional data](tutorials/applications/dask); provide [“layers”](tutorials/start_index) to simultaneously visualize images, models, and analysis results; and easy manual, interactive annotation in 3D.
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]:
```suggestion
This tutorial uses napari 0.4.13.
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]: You can use an internal link here to the API docs:
```suggestion
Along the way, you will see how to access napari functions from [Python code](api/index) and from GUI - though for different purposes, one method might be easier than another. This quick start guide will not cover ALL possible methods but only some ways to perform basic tasks. For the more complete guide, please visit [napari.org](https://napari.org/).
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]:
```suggestion
If you run into any issues, please visit the more detailed [installation guide](https://napari.org/tutorials/fundamentals/installation.html), or [report an issue on GitHub](https://github.com/napari/napari/issues/new/choose)!
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]: If you want to use internal links, you can do
```suggestion
To change the image display through the {class}`~napari.layers.Image` API, in IPython console, type
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]:
```suggestion
![manual_label](./images/manual_label.gif)
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]:
```suggestion
- If you've developed some functionality that you think would benefit the community as a whole, consider publish it as a [plugin](plugins/index)!
```

File: docs/quick_start.md
Comment (melissawm) [inline docs/quick_start.md]:
```suggestion
- Help the napari project by [filing issues](https://github.com/napari/napari/issues) for bugs you encounter or features that you'd like to see! Or even better, come [join the community meeting](community/meeting_schedule) and get to know the team.
```

**Review by melissawm — COMMENTED** [review]
Hi @chili-chiu - the content looks great! I left a few minor comments mostly about internal links, nothing too importante :)

**Comment by sofroniewn** [comment]
Will be great to get this in! I think now that we've got some of the major content reorg PRs in and git-lfs enabled the path should be clear for this one next!!

File: docs/quick_start.md
Comment (evamaxfield) [reply docs/quick_start.md]: Update: this should now be good to go. napari-aicsimageio v0.5.0 has been released that fixes a lot of stuff

**Comment by chili-chiu** [comment]
@DragaDoncila could you help to check if images and internal links are working properly? 
All comments should be addressed by now, thank you!

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
Along the way, you will see how to access napari functions from [Python code](api/index) and from GUI - though for different purposes, one method might be easier than another. This quick start guide will not cover ALL possible methods but only some ways to perform basic tasks. For the more complete guide, please visit [our usage guide](.../usage.md).
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
<img src=".../images/IPython.png" alt="IPython console" width="30%"/>
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
![manual_label](.../images/manual_label.gif)
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
<img src=".../images/ndisplay.png" alt="ndisplay" width="30%"/><br> 
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
To change the image display through the [API](.../api/index), in IPython console, type
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
Several plugins can perform automatic segmentation that takes image layers as input and generates labels layers as output.
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
To analyze labels our layer, we can use [scikit-image](https://scikit-image.org/), a popular Python library that comes with your napari installation. [skimage.measure.regionprops](https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.regionprops) provides a good set of features that can be extracted from labels, including area measurement.
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
:tags: [remove-cell]
```

We can still see a little `1` being printed if we only remove the input, so we should remove the whole cell here (output included).

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
**Note:** the area reported by `regionprops` is the number of pixels. Check pixel size and convert the reported number to physical units.
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
- If you've developed some functionality that you think would benefit the community as a whole, consider publishing it as a [plugin](plugins/index)!
```

File: docs/tutorials/fundamentals/quick_start.md
Comment (DragaDoncila) [inline docs/tutorials/fundamentals/quick_start.md]:
```suggestion
- Help the napari project by [filing issues](https://github.com/napari/napari/issues) for bugs you encounter or features that you'd like to see! Or even better, come [join the community meetings](community/meeting_schedule) and get to know the team.
```

**Review by DragaDoncila — CHANGES_REQUESTED** [review]
Hey @chili-chiu the image links were not quite right, but I've made suggestions to update them (and the internal ones) to what I think they should be. I've also made a few other small suggestions.

~~I would make these updates myself but I don't think I can...? Or at least I don't know how...~~ 🤣  I think I can just commit them so I'll do that and check the build artifact when it's done.

**Review by DragaDoncila — APPROVED** [review]
Okay this looks good to go now! Thanks @chili-chiu 🎉

**Comment by sofroniewn** [comment]
hi @chili-chiu - thanks so much for cleaning this up and thanks @DragaDoncila for reviewing

I see right now that you have merge privileges on the repo and merged this, but I should have made clear that we ask that only napari core developers (see [list here](https://napari.org/community/team.html#current-core-developers)) merge anything into main. That's my bad for not explaining that more clearly.

There's no harm done here though, so don't worry!! These changes aren't even live on the website right now - and it's very easy to make a follow up PR if we want to change anything. Overall having the quick start ready will be great.

Also it has been on my todo list to tighten up the merge permissions so as to avoid this in the future (you're not the first person to have done this). We used to be more limited by zenhub permission issues, but we have more flexibility now that we'll be moving to github projects. I'll get on this asap.

As to the PR itself, I just gave it a read, overall it looks great - the one thing that I might have flagged is if we're ready to promote the `0.4.14` bundle as the first recommended way of installing in the quick-start or if we should wait until the bundle supports update in place. @jni might be interested in reviewing this too, with particular attention to that point. I'm curious what he thinks. As said above we can always make adjustments in a follow up PR!!

**Pull request merged**
